### PR TITLE
Test account sync resilience when no network

### PR DESCRIPTION
**Pull request by gre**
add a test to ensure a sync fails when no network. For instance, a networking error should not yield empty accounts but instead should fail the sync (it was a regression)

### CODE REVIEW

**Comment by gre** [comment]
This works locally but not on CI.. on CI it seems to just times out..

**Comment by juan-cortes** [comment]
I'm taking a look locally and it also passes, I'll update if I find out more about what's happening

**Comment by gre** [comment]
Ok, i just recalled what the issue is:

two **Jest** tests that uses **libcore** makes one of the test **hanging forever**. I never managed to understand why, but it sounds related to junky behavior of Jest when forking test engines (diff processes) and/or probably the behavior of libcore with concurrency.

**Comment by gre** [comment]
I don't know an easy fix. will create a ticket later.

File: src/countervalues/logic.test.js [hunk 1]
```diff
@@ -28,7 +40,39 @@ describe("API sanity", () => {
...
+      calculate(state, { from: bitcoin, to: usd, value: 10000000 })
+    ).toBeGreaterThan(1000);
   });
+  test("resilience when no network with a cache", async () => {
+    const userSettings = {
+      trackingPairs: [
+        {
+          from: bitcoin,
+          to: usd,
+          startDate: new Date(now - 200 * 24 * 60 * 60 * 1000),
```
Comment (juan-cortes) [inline src/countervalues/logic.test.js:53]: Is this arbitrary?

**Review by juan-cortes — APPROVED** [review]

File: src/countervalues/logic.test.js (hunk 1 above)
Comment (gre) [reply src/countervalues/logic.test.js:53]: yeah. like many tests. it's going to ask 200 days of data like if account was 200 days old.

**Comment by gre** [comment]
lol the test is showing to me that if you do a sync with libcore and you did it in the past, it actually succeed 🤯

**Comment by juan-cortes** [comment]
> lol the test is showing to me that if you do a sync with libcore and you did it in the past, it actually succeed 🤯
> because the test taken individually, proving that no network=sync fails, actually succeed. but fails if there were sync before.

image

**Review by valpinkman — APPROVED** [review]

**Pull request closed without merging**